src/main/java/htsjdk/samtools/util/htsget/HtsgetResponse.java

```diff
@@ -0,0 +1,174 @@
+                        conn.setRequestMethod("GET");
+                        this.headers.forEach(conn::setRequestProperty);
+                        conn.connect();
+                        return conn.getInputStream();
+                    } catch (final IOException e) {
+                        throw new RuntimeIOException("Could not retrieve data from block", e);
+                    }
+                case "data":
+                    final String dataUri = this.uri.toString();
+                    if (!dataUri.matches("^data:.*;base64,.*")) {
+                        throw new IllegalArgumentException("data URI must be base64 encoded: " + dataUri);
+                    }
+                    return new ByteArrayInputStream(
+                        Base64.getDecoder().decode(dataUri.replaceFirst("^data:.*;base64,", "")));
+                default:
+                    throw new IllegalArgumentException("Unrecognized URI scheme in data block: " + this.uri.getScheme());
+            }
+        }
+
+        /**
+         * Parse a single data block from a json value
+         *
+         * @param blockJson json value representing a block
+         * @return parsed block object
+         */
+        public static Block parse(final mjson.Json blockJson) {
+            final Block block = new Block();
+
+            final mjson.Json uri = blockJson.at("url");
+            if (uri == null) {
+                throw new IllegalArgumentException("No URI found in Htsget data block");
+            }
+            try {
+                block.uri = new URI(uri.asString());
+            } catch (final URISyntaxException e) {
+                throw new IllegalArgumentException("Failed to parse URI in Htsget data block", e);
+            }
+
+            final mjson.Json dataClass = blockJson.at("class");
+            if (dataClass != null) {
+                block.dataClass = HtsgetClass.valueOf(dataClass.asString().toLowerCase());
+            }
+
+            final mjson.Json headers = blockJson.at("headers");
+            if (headers != null) {
+                block.headers = headers.asJsonMap().entrySet().stream()
+                    .collect(Collectors.toMap(
+                        Map.Entry::getKey,
+                        e -> e.getValue().asString()
+                    ));
+            }
+
+            return block;
+        }
+    }
+
+    private HtsgetFormat format;
+
+    private List<Block> blocks;
+
+    private String md5;
+
+    public HtsgetFormat getFormat() {
+        return this.format;
+    }
+
+    public List<Block> getBlocks() {
+        return Collections.unmodifiableList(this.blocks);
+    }
+
+    public String getMd5() {
+        return this.md5;
+    }
+
+    /**
+     * Parses HtsgetResponse object from json string
+     *
+     * @param s json string
+     * @return parsed HtsgetResponse object
+     */
+    public static HtsgetResponse parse(final String s) {
```

Comment:
We should be checking the API version header value IF we get it back and warning if it's not compatible with our version.

Reply:
Added a check for this